src/Catalyst.Core.Modules.Dfs/CoreApi/UnixFsApi.cs

```diff
@@ -0,0 +1,408 @@
+        }
+
+        public async Task<Stream> ReadFileAsync(string path,
+            long offset,
+            long count = 0,
+            CancellationToken cancel = default)
+        {
+            var stream = await ReadFileAsync(path, cancel).ConfigureAwait(false);
+            return new SlicedStream(stream, offset, count);
+        }
+
+        public async Task<Stream> GetAsync(string path,
+            bool compress = false,
+            CancellationToken cancel = default)
+        {
+            var r = await _nameApi.ResolveAsync(path, true, false, cancel).ConfigureAwait(false);
+            var cid = Cid.Decode(r.Remove(0, 6));
+            var ms = new MemoryStream();
+            await using (var tarStream = new TarOutputStream(ms, 1))
+            {
+                using (var archive = TarArchive.CreateOutputTarArchive(tarStream))
+                {
+                    archive.IsStreamOwner = false;
+                    await AddTarNodeAsync(cid, cid.Encode(), tarStream, cancel).ConfigureAwait(false);
+                }
+            }
+
+            ms.Position = 0;
+            return ms;
+        }
+
+        private async Task AddTarNodeAsync(Cid cid, string name, TarOutputStream tar, CancellationToken cancel)
+        {
+            var block = await _blockApi.GetAsync(cid, cancel).ConfigureAwait(false);
+            var dm = new DataMessage
+            {
+                Type = DataType.Raw
+            };
+            DagNode dag = null;
+
+            if (cid.ContentType == "dag-pb")
+            {
+                dag = new DagNode(block.DataStream);
+                dm = Serializer.Deserialize<DataMessage>(dag.DataStream);
+            }
+
+            var entry = new TarEntry(new TarHeader());
+            var header = entry.TarHeader;
+            header.Mode = 0x1ff; // 777 in octal
+            header.LinkName = string.Empty;
+            header.UserName = string.Empty;
+            header.GroupName = string.Empty;
+            header.Version = "00";
+            header.Name = name;
+            header.DevMajor = 0;
+            header.DevMinor = 0;
+            header.UserId = 0;
+            header.GroupId = 0;
+            header.ModTime = DateTime.Now;
+
+            if (dm.Type == DataType.Directory)
+            {
+                header.TypeFlag = TarHeader.LF_DIR;
+                header.Size = 0;
+                tar.PutNextEntry(entry);
+                tar.CloseEntry();
+            }
+            else // Must be a file
+            {
+                var content = await ReadFileAsync(cid, cancel).ConfigureAwait(false);
+                header.TypeFlag = TarHeader.LF_NORMAL;
+                header.Size = content.Length;
+                tar.PutNextEntry(entry);
+                await content.CopyToAsync(tar, cancel);
+                tar.CloseEntry();
+            }
+
+            // Recurse over files and subdirectories
+            if (dm.Type == DataType.Directory)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This condition was just checked on line 347.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)